test/e2e/upgrades/apps/deployments.go

```diff
@@ -98,89 +86,82 @@ func (t *DeploymentUpgradeTest) Setup(f *framework.Framework) {
 	})
 	framework.ExpectNoError(err)
 
-	// Use observedGeneration to determine if the controller noticed the pod template update.
-	framework.Logf("Wait deployment %q to be observed by the deployment controller", deploymentName)
-	framework.ExpectNoError(framework.WaitForObservedDeployment(c, ns, deploymentName, deployment.Generation))
-
-	// Wait for it to be updated to revision 2
-	By(fmt.Sprintf("Waiting deployment %q to be updated to revision 2", deploymentName))
-	framework.ExpectNoError(framework.WaitForDeploymentRevisionAndImage(c, ns, deploymentName, "2", nginxImage))
-
 	By(fmt.Sprintf("Waiting deployment %q to complete", deploymentName))
-	framework.ExpectNoError(framework.WaitForDeploymentStatus(c, deployment))
+	framework.ExpectNoError(framework.WaitForDeploymentCompletes(c, deployment))
 
-	rs, err = deploymentutil.GetNewReplicaSet(deployment, c.ExtensionsV1beta1())
+	By(fmt.Sprintf("Getting replicaset of deployment %q after rollout", deploymentName))
+	rsList, err = rsClient.List(metav1.ListOptions{LabelSelector: rsSelector.String()})
 	framework.ExpectNoError(err)
-	if rs == nil {
-		framework.ExpectNoError(fmt.Errorf("expected a new replica set for deployment %q", deployment.Name))
+	rss = rsList.Items
+	if len(rss) != 2 {
+		framework.ExpectNoError(fmt.Errorf("expected 2 replicaset, got %d", len(rss)))
 	}
 
-	if rs.UID == t.oldRS.UID {
-		framework.ExpectNoError(fmt.Errorf("expected a new replica set different from the previous one"))
+	By(fmt.Sprintf("Checking replicaset of deployment %q that is created before rollout survives the rollout", deploymentName))
+	// While checking whether old RS survives rollout, we also set t.newRSUID accordingly.
+	if rss[0].UID == oldRSUID {
```

Comment:
nit: use switch-case instead

Reply:
Is it the norm when we have >= 3 if-else cases? Switch case requires 1 more line (that's why i prefer to use switch case when there are 4 if-else cases).

Reply:
In golang, it is idiomatic to write an if-else-if chain as a switch